### PR TITLE
[SYCL] Add new Intel GPU architectures to sycl_ext_oneapi_device_architecture (April 2024)

### PR DESCRIPTION
**Pull request by dm-vodopyanov**
This patch updates sycl_ext_oneapi_device_architecture extension in the following way:

- adds support of new Intel GPU architectures:
  - Meteor Lake U/S and Arrow Lake U/S
  - Meteor Lake H
  - Arrow Lake H
  - Battlemage G21
  - Lunar Lake

- adds missing Intel GPU architecture aliases:
  - `intel_gpu_icl` (== `intel_gpu_icllp`)
  - `intel_gpu_tgl` (== `intel_gpu_tgllp`)
  - aliases "with numbers" [like `intel_gpu_0_0_0`] for `intel_gpu_ehl`, `intel_gpu_rkl`, `intel_gpu_adl_s/p/n`, `intel_gpu_acm_g10/g11/g12`, `intel_gpu_pvc[_vg]`

- adds minor NFC changes to the source code

### CODE REVIEW

**Comment by dm-vodopyanov** [comment]
@sergey-semenov @intel/llvm-reviewers-runtime @intel/dpcpp-clang-driver-reviewers @intel/dpcpp-specification-reviewers @intel/dpcpp-doc-reviewers - friendly ping

**Review by AlexeySachkov — APPROVED** [review]

**Review by sergey-semenov — APPROVED** [review]

File: clang/lib/Driver/ToolChains/SYCL.cpp [hunk 1]
```diff
@@ -896,19 +896,31 @@ StringRef SYCL::gen::resolveGenDevice(StringRef DeviceName) {
...
+          .Cases("intel_gpu_acm_g12", "intel_gpu_dg2_g12", "intel_gpu_12_57_0",
+                 "acm_g12")
+          .Cases("intel_gpu_pvc", "intel_gpu_12_60_7", "pvc")
+          .Cases("intel_gpu_pvc_vg", "intel_gpu_12_61_7", "pvc_vg")
+          .Cases("intel_gpu_mtl_u", "intel_gpu_mtl_s", "intel_gpu_arl_u",
+                 "intel_gpu_arl_s", "intel_gpu_12_70_4", "mtl_u")
+          .Cases("intel_gpu_mtl_h", "intel_gpu_12_71_4", "mtl_h")
+          .Cases("intel_gpu_arl_h", "intel_gpu_12_74_4", "arl_h")
+          .Cases("intel_gpu_bmg_g21", "intel_gpu_20_1_4", "bmg_g21")
+          .Cases("intel_gpu_lnl_m", "intel_gpu_20_4_4", "lnl_m")
```
Comment (mdtoguchi) [inline clang/lib/Driver/ToolChains/SYCL.cpp:923]: Could you add some test entries for the new values in `clang/test/Driver/sycl-oneapi-gpu-intelgpu.cpp` ?

Comment (dm-vodopyanov) [reply clang/lib/Driver/ToolChains/SYCL.cpp:923]: Thanks, added in https://github.com/intel/llvm/pull/13520/commits/b9492ceaf7f17bbd951c2f0651e71c4a1226830e

**Review by mdtoguchi — APPROVED** [review]
Thanks, LGTM

**Comment by dm-vodopyanov** [comment]
@gmlueck could you please take a look if you have some time?

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc [hunk 2]
```diff
@@ -161,15 +161,15 @@ a|
...
+|Intel(R) microarchitecture code name Broadwell Intel graphics architecture.
 
 a|
 [source]
 ----
 intel_gpu_skl
 ----
 |-
-|Skylake Intel graphics architecture.
+|Intel(R) microarchitecture code name Skylake Intel graphics architecture.
```
Comment (gmlueck) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc:172]: Is there a reason not to add this same wording to all the other architecture names below?

Comment (dm-vodopyanov) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc:172]: In short, yes, answered internally

**Review by gmlueck — APPROVED** [review]

**Pull request merged**